contracts/mixnet/src/storage.rs

```diff
@@ -46,6 +47,32 @@ pub fn config_read(storage: &dyn Storage) -> ReadonlySingleton<State> {
     singleton_read(storage, CONFIG_KEY)
 }
 
+pub fn total_mix_stake(storage: &dyn Storage) -> ReadonlySingleton<Uint128> {
+    singleton_read(storage, TOTAL_STAKE_KEY)
+}
+
+pub fn mut_total_mix_stake(storage: &mut dyn Storage) -> Singleton<Uint128> {
+    singleton(storage, TOTAL_STAKE_KEY)
+}
+
+pub fn incr_total_mix_stake(amount: Uint128, storage: &mut dyn Storage) {
+    let stake = total_mix_stake(storage)
+        .load()
+        .unwrap()
+        .checked_add(amount)
```

Comment:
If we're doing a checked_add then we shouldn't be really unwraping it, should we? Either deal with failure cases or just change it into `wrapping_add` or `saturating_add`

Reply:
This is a good point, we likely want saturating add anyway

Reply:
Altough saturating has the potential of leading to weirdness, probably better to just handle the error